File: backend/libs/store/src/objects/search_repository.rs
use chrono::NaiveDateTime;
use sea_orm::{prelude::Uuid, FromQueryResult};

#[derive(FromQueryResult, Serialize, Deserialize, ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct SearchRepository {
    pub id: Uuid,
    pub repository_name: String,
    pub language: Option<String>,
    pub stargazers_count: i64,
    pub forks_count: i64,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
    pub archived: bool,
    pub project_id: Uuid,
    pub project: String,
    pub url: String,
}

File: frontend/src/types/searchRepository.ts
export interface SearchRepository {
    id: string;
    repositoryName: string;
    language?: string;
    stargazersCount: number;
    forksCount: number;
    createdAt: Date;
    updatedAt: Date;
    archived: boolean;
    projectId: string;
    project: string;
    url: string;
}

export interface SearchRepositoryParams {
    orderBy?: string;
    order?: string;
    page?: number;
    perPage?: number;
    language?: string;
    repository?: string;
    projectId?: string;
    project?: string;
    archived?: boolean;
    fork?: boolean;
}